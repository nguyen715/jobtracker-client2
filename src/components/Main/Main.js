import React, { useContext } from 'react';
import { Route } from 'react-router-dom';
import './Main.css';
import Context from '../../context/Context';
import Info from '../Info/Info.js';
import Welcome from '../Welcome/Welcome.js';
import PostList from '../PostList/PostList.js';
import NewPostForm from '../NewPostForm/NewPostForm.js';

function Main() {
  const context = useContext(Context);

  return (
    <div className="main">
      <Route path="/welcome" component={Welcome}></Route>
      <Route path="/info" component={Info}></Route>
      <Route path="/jobs" component={PostList}></Route>
      <Route path="/create-post" component={NewPostForm}></Route>
    </div>
  )
};

export default Main;