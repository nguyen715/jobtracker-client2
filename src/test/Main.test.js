import React, { useContext } from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Context from '../context/Context';

import Main from '../components/Main/Main.js';

// const context = useContext(Context);

describe('Main component', () => {
  it('renders the UI as expected', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Main />
      </BrowserRouter>
    )
    .toJSON()

    expect(tree).toMatchSnapshot()
  })
})