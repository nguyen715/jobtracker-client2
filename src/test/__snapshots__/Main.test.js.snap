// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders the UI as expected 1`] = `
<div
  className="main"
>
  <div
    className="welcome"
  >
    <p>
      Welcome to Jobtracker App! Use this app to keep track of job listings that you find during your job search. Enter your email address below to get started, and use the same email to come back to your saved list at any time.
    </p>
    <p>
      Happy job hunting!
    </p>
    <form
      action=""
      onSubmit={[Function]}
    >
      <label
        htmlFor="email"
      >
        Enter your email address: 
      </label>
      <input
        id="email"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <div
        className="btn-div"
      >
        <button
          type="submit"
        >
          View My Saved Jobs
        </button>
      </div>
    </form>
  </div>
</div>
`;
